daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -345,6 +347,63 @@ allocateParty args name = do
         setRequestBodyJSON $ AllocatePartyRequest {identifierHint = text, displayName = text}
   return party
 
+runLedgerReset :: LedgerFlags -> IO ()
+runLedgerReset flags = do
+  putStrLn "Resetting ledger."
+  args <- getDefaultArgs flags
+  reset args
+
+reset :: LedgerArgs -> IO ()
+reset args =
+  case api args of
+    Grpc ->
+      runWithLedgerArgs args $ do
+        parties <- map L.party <$> L.listKnownParties
+        unless (null parties) $ do
+          ledgerId <- L.getLedgerIdentity
+          L.setToken
+            (L.Token $
+             T.unpack $
+             tokenFor
+               [TL.toStrict $ L.unParty p | p <- parties]
+               (TL.toStrict $ L.unLedgerId ledgerId)
+               "ledger-reset") $ do
+            activeContracts <-
+              L.getActiveContracts
+                ledgerId
+                (TransactionFilter $
+                 Map.fromList [(L.unParty p, Just $ Filters Nothing) | p <- parties])
+                (L.Verbosity False)
+            let cmds =
+                  L.Commands
+                    { coms =
+                        [ L.ExerciseCommand
+                          { tid = tid
+                          , cid = cid
+                          , choice = L.Choice "Archive"
+                          , arg = L.VRecord $ L.Record Nothing []
+                          }
+                        | (_offset, _mbWid, events) <- activeContracts
+                        , L.CreatedEvent {cid, tid} <- events
+                        ]
+                    , lid = ledgerId
+                    , wid = Nothing
+                    , aid = L.ApplicationId "hotreload"
+                    , cid = L.CommandId "reset"
+                    , actAs = parties
+                    , readAs = parties
```

Comment:
you don’t need `readAs`. `actAs` always implies `readAs`.

Reply:
ok, I'll remove it.